### PR TITLE
Add contributing docs

### PR DESCRIPTION
**Pull request by twpayne**
cc @dschep

### CODE REVIEW

**Comment by dschep** [comment]
This is fantastic. Thanks!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# Contributing
+
+`go-geos` uses `libgeos`'s stable C API,
+[`geos_c.h`](http://libgeos.org/doxygen/geos__c_8h.html). Only the thread-safe
+`*_r` functions are used.
+
+## Adding methods to `*Geom`
+
+Wherever possible, `go-geom` uses code generation to generate wrappers for
```
Comment (dschep) [inline CONTRIBUTING.md:9]: Shouldn't this read `go-geos`?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  contains the high-level definitions of the methods.
+* [`geommethods.go.tmpl`](https://github.com/twpayne/go-geos/blob/master/geommethods.go.tmpl)
+  is a `text/template` template that is executed with the data from
+  `geommethods.yaml`.
+* [`internal/cmds/execute-template/`](https://github.com/twpayne/go-geos/tree/master/internal/cmds/execute-template)
+  executes a template with data and includes custom template functions.
+* [`go generate`](https://go.dev/blog/generate) runs
+  `internal/cmds/execute-template/` with `geommethods.yaml` and
+  `geommethods.go.tmpl` as inputs and writes `geommethods.go`.
+* [`geom_test.go`](https://github.com/twpayne/go-geos/blob/master/geos_test.go)
```
Comment (dschep) [inline CONTRIBUTING.md:25]: This is a link to https://github.com/twpayne/go-geos/blob/master/geos_test.go but should be a link to https://github.com/twpayne/go-geos/blob/master/geom_test.go

**Comment by twpayne** [comment]
Thanks for catching these :) Fixed in ad46ef6b9c632c19e1a3f4dac922c8fcb7c0c5b0.

**Pull request merged**